### DISCUSSION TITLE
Meta data from text file

### DISCUSSION THREAD

**Discussion by Woisek** [vladmandic/sdnext#694] [Ideas]
Just now, I was thinking about the meta data generated and stored in the PNGs. We also have the ability to save those data as a text file. When an image gets lost somehow, it can't be recreated. But if a meta data text file exists, it can be restored.
Now I thought, it would be convenient, if we also could drag such text files into the place where we would put images, and it would readout this text file and fill in the meta data from it.
I know we could do this by hand, but ... you know ... typing errors and laziness ... 😁

Would this be thinkable to do? 😀

**Comment by tkiikt** [comment]
That's something to do with EXTENSION or stand-alone software. That method is not easy because there is a high possibility that different data will be written by mistake and you have to manage file correspondence.
Why not just use [this](https://www.reddit.com/r/StableDiffusion/comments/130r2jf/stable_diffusion_prompt_reader_v12/) software? Oops, perhaps you are making a request? Let's wait patiently.😁

**Reply by Woisek** [reply]
>That method is not easy because there is a high possibility that different data will be written by mistake and you have to manage file correspondence.

A "different data" could also be in a PNG (maybe another SD program wrote it). Not likely, but still possible. The only difference I see is the kind of reading the meta data done on PNG and txt files. Other than that, nothing more exciting should be happening. If something is off, it simply wouldn't work or at least insert only partial data into WebUI (with a hint, that only partial data was used).

>Why not just use [this](https://www.reddit.com/r/StableDiffusion/comments/130r2jf/stable_diffusion_prompt_reader_v12/) software?

I saw this software, it's cool, but lacks the important feature to _write_ meta data into PNGs on our own. But it should be coming, as the developer said to me.
But besides that, this isn't what I suggested. This software also uses images for the readout. I'm talking about the _text files_, that are generated besides the generated images (if this option is checked).

**Comment by vladmandic** [comment]
just to be clear, the ask is to automatically write txt file next to generated image file and ability to read that text file using process image tab?

**Reply by Woisek** [reply]
>the ask is to automatically write txt file next to generated image file 

Well, this option already exists. One has only to check it if needed.

>ability to read that text file using process image tab

Exactly this! Like I said, if somehow a PNG is lost, so is the saved meta data in it. But if a text file exists, instead of manually copy pasting all data from it, it would be convenient to just drag and drop it like a original PNG.
At least I think it would be helpful if this would be possible (and because it happened to me also). 🙂
Aaand, came just into my mind, very convenient for people who share data. So instead of sending big sized PNGs for the meta data, small text files could be used.

**Comment by vladmandic** [comment]
done. 
- enable creating of text file in settings. 
- drag a text file to prompt text area
- click read generation params (first small button on the right)

**Reply by Woisek** [reply]
Well ... that was fast .. 😱 ... will check later on. Thanks! 👍

**Reply by Woisek** [reply]
Working great! Thanks again. 👍
